pandas/io/formats/style.py

```diff
@@ -322,6 +322,10 @@ def set_tooltips(
             raise NotImplementedError(
                 "Tooltips can only render with 'cell_ids' is True."
             )
+        if not ttips.index.is_unique or not ttips.columns.is_unique:
+            raise KeyError(
```

Comment:
all these have tests?

Reply:
`def test_tooltips_non_unique_raises(styler)` and `def test_set_td_classes_non_unique_raises(styler)`